File: ai25btech11021/matgeo/4.13.90/codes/main.py
import numpy as np
import matplotlib.pyplot as plt
from mpl_toolkits.mplot3d import Axes3D

P = np.array([1, -2, 1])
F = np.array([8/3, 4/3, -7/3])
A, B, C, D_plane = 1, 2, -2, 10

fig = plt.figure(figsize=(10, 8))
ax = fig.add_subplot(111, projection='3d')

ax.scatter(*P, color='r', s=100, label=r'Point P(1, -2, 1)')
ax.scatter(*F, color='g', s=100, label=r'Foot of Perpendicular F($\frac{8}{3}, \frac{4}{3}, -\frac{7}{3}$)')

ax.plot([P[0], F[0]], [P[1], F[1]], [P[2], F[2]], 'k--', label='Perpendicular Line PF')

x_min, x_max = -1, 4
y_min, y_max = -4, 4

xx, yy = np.meshgrid(np.linspace(x_min, x_max, 10), np.linspace(y_min, y_max, 10))

zz = (A * xx + B * yy - D_plane) / C

ax.plot_surface(xx, yy, zz, alpha=0.5, rstride=100, cstride=100, color='c')

ax.set_xlabel('X-axis')
ax.set_ylabel('Y-axis')
ax.set_zlabel('Z-axis')
ax.set_title(r'3D Plot of Point, Plane, and Foot of Perpendicular')

ax.legend()
ax.view_init(elev=20, azim=45)
ax.set_box_aspect([np.ptp(a) for a in [ax.get_xlim(), ax.get_ylim(), ax.get_zlim()]]) 

plt.savefig('3d_point_plane_perpendicular.png')

File: ai25btech11021/matgeo/4.13.90/codes/main.c
#include <stdio.h>
#include <math.h>

double distanceFromPointToPlane(double A, double B, double C, double D_plane,
                                double x0, double y0, double z0) {
    double numerator = fabs(A * x0 + B * y0 + C * z0 - D_plane);
    double denominator = sqrt(A * A + B * B + C * C);
    return numerator / denominator;
}

int main() {
    double A = 1, B = 2, C = -2;
    double alpha = 10;
    double x0 = 1, y0 = -2, z0 = 1;
    double distance = distanceFromPointToPlane(A, B, C, alpha, x0, y0, z0);
    printf("Distance from point to plane = %.2f\n", distance);
    return 0;
}
